engine/src/main/java/org/terasology/input/InputSystem.java

```diff
@@ -151,7 +151,14 @@ public boolean isCapturingMouse() {
      * Updates the client and input entities of the local player, to be used in input events against the local player.
      */
     private void updateInputEntities() {
-        inputEntities = new EntityRef[] {localPlayer.getClientEntity(), localPlayer.getCharacterEntity()};
+        if (inputEntities == null
+                || inputEntities.length != 2
+                || inputEntities[0] == null
+                || inputEntities[1] == null
+                || !inputEntities[0].equals(localPlayer.getClientEntity())
+                || !inputEntities[1].equals(localPlayer.getCharacterEntity())) {
+            inputEntities = new EntityRef[]{localPlayer.getClientEntity(), localPlayer.getCharacterEntity()};
+        }
```

Comment:
Is there a specific reason why you avoid modifying the `inputEntities` array? The name of the method seems to imply that you always update the array, as was done before.

Reply:
It was modified to solve Issue #2949 - to reduce unnecessary creations of new `inputEntities` array.
Also, the name of the method (`updateInputEntities()`) is slightly misleading - it's not only updates array, it's initialize it on first call.

Reply:
Ah, I see, thanks for clearing that up. In that case, do you mind adding a "Closes #2949" or "Resolves #2949" to the PR description? It will ensure that GitHub [magically closes that issue when this PR is merged](https://help.github.com/en/articles/closing-issues-using-keywords) and let PR reviewers know that this PR aims to fix that issue as well!

Reply:
Ok, I'll do

Reply:
Noob question - how to change PR description?

Reply:
@AahzBrut sorry I didn't see this earlier -- you can edit a comment or an issue/PR description by clicking on the 3-dot menu on the top right corner and clicking "Edit". Since this PR has already been merged, I've closed the issue. Thanks for your contribution!